skfp/fingerprints/atom_pair.py

```diff
@@ -206,14 +215,31 @@ def _calculate_fingerprint(
             countSimulation=self.count_simulation,
         )
         if self.count:
-            X = [
-                gen.GetCountFingerprintAsNumPy(mol, confId=conf_id)
-                for mol, conf_id in zip(X, conf_ids)
-            ]
+            if self.normalize:
+                X = [
+                    self._scale_by_hac(
+                        gen.GetCountFingerprintAsNumPy(mol, confId=conf_id), mol
+                    )
+                    for mol, conf_id in zip(X, conf_ids)
+                ]
+            else:
+                X = [
+                    gen.GetCountFingerprintAsNumPy(mol, confId=conf_id)
+                    for mol, conf_id in zip(X, conf_ids)
+                ]
         else:
             X = [
                 gen.GetFingerprintAsNumPy(mol, confId=conf_id)
                 for mol, conf_id in zip(X, conf_ids)
             ]
+            if self.normalize:
+                warnings.warn("Scaling by HAC can only be applied to count vectors.")
 
         return csr_array(X) if self.sparse else np.array(X)
+
+    def _scale_by_hac(self, fingerprint: np.ndarray, mol: Mol) -> np.ndarray:
+        """Scale computed fingerprint by the heavy atom count (HAC).
+
+        Vector values are expressed in percent and rounded to the nearest integer.
+        """
```

Comment:
If this is a private method, maybe just leave a regular comment? Nobody should be interested in the docstring anyway.
```
# scale values to percentages, rounded to the nearest integer
```